generator/shared.go

```diff
@@ -247,7 +247,7 @@ func DefaultSectionOpts(gen *GenOpts, client bool) {
 	}
 
 	if len(sec.Operations) == 0 {
-		if client {
+		if gen.EnsureDefaultOpts {
```

Comment:
this variable name is very misleading because this is about picking a set of templates for either client or server gen.
I don't know why EnsureDefaultOpts should be the switch when it's already doing the default opts.